### ISSUE TITLE
[Tesla Binding]: Tesla Account Handler Acces token refresh not working

### ISSUE THREAD

**Issue by hlemmens** [openhab/openhab-addons#16382]
I run Openhab 4.2.0-SNAPSHOT - Build #3879 with Tesla binding 4.1.2.202401252209

It seems the new snapshot is unable to refresh the access token:

2024-02-05 06:19:40.318 [DEBUG] [internal.handler.TeslaVehicleHandler] - Querying the vehicle, response : 401, Unauthorized
2024-02-05 06:19:40.319 [DEBUG] [internal.handler.TeslaVehicleHandler] - The access token has expired, trying to get a new one.
2024-02-05 06:19:40.320 [DEBUG] [internal.handler.TeslaAccountHandler] - Current authentication time 2024-02-05 06:19:40
2024-02-05 06:19:40.321 [DEBUG] [internal.handler.TeslaAccountHandler] - Found a request token from 2024-02-04 22:18:47
2024-02-05 06:19:40.322 [DEBUG] [internal.handler.TeslaAccountHandler] - Access token expiration time 2024-02-05 06:18:47
2024-02-05 06:19:40.322 [DEBUG] [internal.handler.TeslaAccountHandler] - The access token has expired
2024-02-05 06:19:40.323 [DEBUG] [sla.internal.handler.TeslaSSOHandler] - Exchanging SSO refresh token for API access token
**2024-02-05 06:19:40.348 [DEBUG] [sla.internal.handler.TeslaSSOHandler] - An exception occurred while invoking a HTTP request: 'java.io.EOFException: HttpConnectionOverHTTP@4df9f0ba::DecryptedEndPoint@1fa09643{l=/192.168.0.13:39004,r=auth.tesla.com/2.21.16.79:443,OPEN,fill=-,flush=-,to=28853028/0}'**
2024-02-05 06:19:40.349 [DEBUG] [sla.internal.handler.TeslaSSOHandler] - An error occurred during refresh of SSO token: no response
2024-02-05 06:19:40.350 [DEBUG] [internal.handler.TeslaVehicleHandler] - An error occurred while querying the vehicle

This is the same issue as we had in https://github.com/openhab/openhab-addons/issues/14435

Anyone able to fix?

## Possible Solution
This is the same issue as we had in https://github.com/openhab/openhab-addons/issues/14435. Can we apply the same fix?

## Steps to Reproduce (for Bugs)
Every 8 hours when the access token expires, the refresh fails.

## Your Environment
Openhab 4.2.0-SNAPSHOT - Build https://github.com/openhab/openhab-addons/pull/3879 with Tesla binding 4.1.2.202401252209

**Comment by hlemmens** [comment]
No reactions so far. Am I the only one with this issue.

No smart guy who can bring a solution? At least I'm not! Please offer some help.

**Comment by psmedley** [comment]
I suspect  this is due to the progressive rollout of the new Fleet  API...

**Comment by kaikreuzer** [comment]
> Openhab 4.2.0-SNAPSHOT - Build https://github.com/openhab/openhab-addons/pull/3879 with Tesla binding 4.1.2.202401252209

You should always run the bindings that come with the snapshot if you expect anyone trying to reproduce it.
I am on the 4.2.0-SNAPSHOT Tesla binding and do not have any such issue.

**Comment by hlemmens** [comment]
I reinstalled Snapshot 4.2.0 with openhabian-config.



The binding doesn’t work at all now. Back to 4.2.0

**Comment by kaikreuzer** [comment]
Looks as if you have this old bundle in your addons folder. Here's what you should see with the latest snapshot:

```
openhab> list -s | grep knx                                                                                                                                    
311 x Active x  80 x 4.2.0.202402250349     x org.openhab.binding.knx
```

**Comment by hlemmens** [comment]
This is what I tried out:

openhab> list -s | grep knx                                                                                                                                                         
openhab> bundle:list -s |grep knx
openhab> bundle:list |grep knx    

No result. It looks lice this .knx file is not there.

**Comment by hlemmens** [comment]
I reinstalled the Snapshot and manually added the knx binding.

Seems to work now. Let’s wait and see if in 8 hours from now the access token will be renewed.

**Comment by digitaldan** [comment]
I don't think @kaikreuzer meant to grep for `knx`, he meant `tesla`, it was probably a copy and paste error from his console history.  The KNX binding doesn't have anything todo with Tesla and is not related.

**Comment by kaikreuzer** [comment]
Yes, clearly @digitaldan is right - I accidentially posted the knx grep, then manually fixed it, but missed to fix the fist line of it. 🙄 .

**Comment by hlemmens** [comment]
This is the result of list -s | grep tesla:
 27 │ Active   │  80 │ 4.1.2.202401252209     │ org.openhab.binding.tesla
It seems to be the right one.

It works for the time being. Will see tomorrow morning if the access token refresh works well.

Best regards,

Hubert.

**Comment by kaikreuzer** [comment]
> It seems to be the right one.

It's not. It is a version 4.1.2 from January. You must have installed that manually, e.g. by placing it in the addons folder. You need to remove it.

**Comment by hlemmens** [comment]
OK, sorry for the confusion.

This is what I found in the latest Snapshot - Adons and installed:
 27 │ Active   │  80 │ 4.2.0.202402260411     │ org.openhab.binding.tesla

I see the date is on day later than the version Kai mentioned. I suppose this is not an issue?

**Comment by kaikreuzer** [comment]
That's indeed the latest build, so all should be fine.

**Comment by hlemmens** [comment]
Unfortunately it isn't: access token  refresh fails:

**2024-02-26 18:18:52.169 [DEBUG] [internal.handler.TeslaVehicleHandler] - Querying the vehicle, response : 401, Unauthorized
2024-02-26 18:18:52.170 [DEBUG] [internal.handler.TeslaVehicleHandler] - The access token has expired, trying to get a new one.
2024-02-26 18:18:52.170 [DEBUG] [internal.handler.TeslaAccountHandler] - Current authentication time 2024-02-26 18:18:52
2024-02-26 18:18:52.171 [DEBUG] [internal.handler.TeslaAccountHandler] - Found a request token from 2024-02-26 10:18:14
2024-02-26 18:18:52.171 [DEBUG] [internal.handler.TeslaAccountHandler] - Access token expiration time 2024-02-26 18:18:14
2024-02-26 18:18:52.171 [DEBUG] [internal.handler.TeslaAccountHandler] - The access token has expired
2024-02-26 18:18:52.171 [DEBUG] [sla.internal.handler.TeslaSSOHandler] - Exchanging SSO refresh token for API access token
2024-02-26 18:18:52.189 [DEBUG] [sla.internal.handler.TeslaSSOHandler] - An exception occurred while invoking a HTTP request: 'java.io.EOFException: HttpConnectionOverHTTP@df984b5::DecryptedEndPoint@50870dcf{l=/192.168.0.13:55528,r=auth.tesla.com/2.21.16.79:443,OPEN,fill=-,flush=-,to=28837365/0}'
2024-02-26 18:18:52.189 [DEBUG] [sla.internal.handler.TeslaSSOHandler] - An error occurred during refresh of SSO token: no response
2024-02-26 18:18:52.190 [DEBUG] [internal.handler.TeslaVehicleHandler] - An error occurred while querying the vehicle

**
